### PR TITLE
Destroyed texture submitting validation test review and implementation

### PR DESCRIPTION
**Pull request by jzm-intel**
Review and implement validation tests on submitting destroyed texture.

Issue: #904

Note: Modified api,validation,texture,destroy:submit_a_destroyed_texture_as_attachment tests `stencil8` texture format.
<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by jzm-intel** [comment]
Currently the relating validation tests are implemented separately:
- writeTexture, copyB2T, copyT2B: api,validation,image_copy,texture_related:valid
- coptT2T: api,validation,encoding,cmds,copyTextureToTexture:copy_with_invalid_or_destroyed_texture
- setBindGroup: api,validation,encoding,cmds,setBindGroup:state_and_binding_index
- copyExternalImageToTexture: api,validation,queue,copyToTexture,CopyExternalImageToTexture:destination_texture,state
- Render attachment: api,validation,texture,destroy:submit_a_destroyed_texture_as_attachment

**Comment by jzm-intel** [comment]
Please take a look, thanks @kainino0x @austinEng

**Review by austinEng — APPROVED** [review]

File: src/webgpu/api/validation/texture/destroy.spec.ts
Comment (jzm-intel) [reply src/webgpu/api/validation/texture/destroy.spec.ts]: Done.

File: src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts
Comment (jzm-intel) [reply src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts:27]: Yes, but existing code in this testsuit call this helper function (`TestCopyTextureToTexture`) a lot, changing it in this way by adding a optional parameter (rather than modifying `isSuccess` into a three values enum) ensure that we can leave the existing code unchanged. On the other hand, many other similar helper function do the same thing, e.g.  `testRun` in `api\validation\image_copy\image_copy.ts`.

Comment (austinEng) [reply src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts]: hm I think it's ok to change the existing code to improve the readability. That way it's more obvious if someone is reading it what the test is expecting to happen

Comment (jzm-intel) [reply src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts]: Done.

Comment (austinEng) [reply src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts]: Thanks, still LGTM!

**Comment by jzm-intel** [comment]
Thanks for your review!

**Comment by kainino0x** [comment]
I don't think this needs a second review; please 'squash and merge' if ready.

**Pull request merged**
